util/src/time.cc

```diff
@@ -0,0 +1,66 @@
+#include <boost/date_time/posix_time/posix_time.hpp>
+#include <leatherman/util/time.hpp>
+
+namespace leatherman { namespace util {
+
+    std::string getExpiryDatetime(int expiry_minutes) {
+        struct tm expiry_time_info;
+        char expiry_time_buffer[80];
+
+        // Get current time and add the specified minutes
+        time_t expiry_time { time(nullptr) };
+        expiry_time += 60 * expiry_minutes;
+
+        // Get local time structure
+        get_local_time(&expiry_time, &expiry_time_info);
+
+        // Return the formatted string
+        if (strftime(expiry_time_buffer, 80, "%FT%TZ", &expiry_time_info) == 0) {
+            // invalid buffer
+            return "";
+        }
+
+        return std::string { expiry_time_buffer };
+    }
+
+    void displayProgress(double percent, int len, std::string status) {
+        printf("%c[2K", 27);
```

Comment:
So, I was able to make this work cross platform by not actually clearing the line, but just printing over it. The way this is implemented, the length seems to be constant, or else only increasing in length (going from 99% to 100%). Can anyone think of any time when this would not be the case?

There is not really a clean way to do it on Windows (messiness involving manually specifying the cursor position or printing a backspace a certain number of times), but we can try one of those if we think the line really needs to be cleared.

As a side note, is this even being used currently anywhere? I couldn't find any references in any of the projects...